plugins/src/main/java/com/google/fhir/proxy/plugin/OpenSRPSyncAccessDecision.java

```diff
@@ -193,4 +206,108 @@ private boolean isResourceTypeRequest(String requestPath) {
 
     return false;
   }
+
+  @VisibleForTesting
+  protected IgnoredResourcesConfig getIgnoredResourcesConfigFileConfiguration(String configFile) {
+    if (configFile != null && !configFile.isEmpty()) {
+      try {
+        Gson gson = new Gson();
+        config = gson.fromJson(new FileReader(configFile), IgnoredResourcesConfig.class);
+        if (config == null || config.entries == null) {
+          throw new IllegalArgumentException("A map with a single `entries` array expected!");
+        }
+        for (IgnoredResourcesConfig entry : config.entries) {
+          if (entry.getPath() == null) {
+            throw new IllegalArgumentException("Allow-list entries should have a path.");
+          }
+        }
+
+      } catch (IOException e) {
+        logger.error("IO error while reading sync-filter skip-list config file {}", configFile);
+      }
+    }
+
+    return config;
+  }
+
+  @VisibleForTesting
+  protected IgnoredResourcesConfig getSkippedResourcesConfigs() {
+    return getIgnoredResourcesConfigFileConfiguration(
+        System.getenv(SYNC_FILTER_IGNORE_RESOURCES_FILE_ENV));
+  }
+
+  /**
+   * This method checks the request to ensure the path, request type and parameters match values in
+   * the hapi_sync_filter_ignored_queries configuration
+   */
+  private boolean shouldSkipDataFiltering(ServletRequestDetails servletRequestDetails) {
+    if (config == null) return false;
+
+    for (IgnoredResourcesConfig entry : config.entries) {
+
+      if (!entry.getPath().equals(servletRequestDetails.getRequestPath())) {
+        continue;
+      }
+
+      if (entry.getMethodType() != null
+          && !entry.getMethodType().equals(servletRequestDetails.getRequestType().name())) {
+        continue;
+      }
+
+      for (Map.Entry<String, Object> expectedParam : entry.getQueryParams().entrySet()) {
+        String[] actualQueryValue =
+            servletRequestDetails.getParameters().get(expectedParam.getKey());
+
+        if (actualQueryValue == null) {
+          return true;
+        }
+
+        if (MATCHES_ANY_VALUE.equals(expectedParam.getValue())) {
+          return true;
+        } else {
+          if (actualQueryValue.length != 1) {
+            // We currently do not support multivalued query params in skip-lists.
+            return false;
+          }
+
+          if (expectedParam.getValue() instanceof List) {
+
+            if (CollectionUtils.isEqualCollection(
+                (List) expectedParam.getValue(), Arrays.asList(actualQueryValue[0].split(",")))) {
+              return true;
+            } else {
+              return false;
```

Comment:
I think we can simplify this check.  `return CollectionUtils.isEqualCollection(
                (List) expectedParam.getValue(), Arrays.asList(actualQueryValue[0].split(",")))`

Reply:
I'm open to ideas. Basically we have two lists and we want to ensure that the values they contain are exactly the same in which ever order.

Reply:
Made the update. Upgraded the tag too

Reply:
Looks good, more concise 👍🏾